spring-boot-actuator/src/main/java/org/springframework/boot/actuate/endpoint/ScheduledTaskEndpoint.java

```diff
@@ -0,0 +1,179 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.endpoint;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonInclude.Include;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.scheduling.annotation.SchedulingConfigurer;
+import org.springframework.scheduling.config.CronTask;
+import org.springframework.scheduling.config.IntervalTask;
+import org.springframework.scheduling.config.ScheduledTaskRegistrar;
+import org.springframework.scheduling.config.Task;
+import org.springframework.scheduling.config.TriggerTask;
+import org.springframework.scheduling.support.ScheduledMethodRunnable;
+
+/**
+ * {@link Endpoint} to expose scheduled task information.
+ *
+ * @author Yunkun Huang
+ * @since 2.0.0
+ */
+@ConfigurationProperties(prefix = "endpoints.schedules")
+public class ScheduledTaskEndpoint
+		extends AbstractEndpoint<List<ScheduledTaskEndpoint.ScheduledTaskInformation>>
+		implements SchedulingConfigurer {
+	private List<ScheduledTaskInformation> list = new ArrayList<>();
+
+	public ScheduledTaskEndpoint() {
+		super("schedules");
+	}
+
+	/**
+	 * Create a new {@link ScheduledTaskEndpoint} instance.
+	 */
+	@Override
+	public List<ScheduledTaskInformation> invoke() {
+		return this.list;
+	}
+
+	@Override
+	public void configureTasks(ScheduledTaskRegistrar taskRegistrar) {
+		this.list.addAll(build(taskRegistrar.getFixedRateTaskList(), ScheduledType.FIXEDRATE));
+		this.list.addAll(build(taskRegistrar.getCronTaskList(), ScheduledType.CRON));
+		this.list.addAll(build(taskRegistrar.getFixedDelayTaskList(), ScheduledType.FIXEDDELAY));
+		this.list.addAll(build(taskRegistrar.getTriggerTaskList(), ScheduledType.TRIGGER));
+	}
+
+	private List<ScheduledTaskInformation> build(final List<? extends Task> tasks, final ScheduledType type) {
+		return tasks.stream().map(task -> {
+			ScheduledTaskInformation scheduledTaskInformation = new ScheduledTaskInformation();
+			scheduledTaskInformation.setType(type);
+			scheduledTaskInformation.setName(getGenericName(task.getRunnable()));
+			if (task instanceof IntervalTask) {
+				IntervalTask intervalTask = (IntervalTask) task;
+				scheduledTaskInformation.setInitialDelay(intervalTask.getInitialDelay());
+				scheduledTaskInformation.setInterval(intervalTask.getInterval());
+			}
+			if (task instanceof CronTask) {
+				CronTask cronTask = (CronTask) task;
+				scheduledTaskInformation.setExpression(cronTask.getExpression());
+			}
+			if (task instanceof TriggerTask) {
+				TriggerTask triggerTask = (TriggerTask) task;
+				scheduledTaskInformation.setTrigger(triggerTask.getTrigger().toString());
+			}
```

Comment:
I suspect that it's impossible that task is the `IntervalTask` and `CronTask` and `TriggerTask` at the same time. In this case, it's better to use if-else if-else if.